server/src/session.ts

```diff
@@ -121,6 +131,22 @@ export class Session {
     conn.onHover(p => this.onHover(p));
     conn.onCompletion(p => this.onCompletion(p));
     conn.onCompletionResolve(p => this.onCompletionResolve(p));
+    conn.onRequest(lspext.GetTcb, p => this.onGetTcb(p));
+  }
+
+  private onGetTcb(params: lspext.GetTcbParams): ngls.GetTcbResponse {
+    const lsInfo = this.getLSAndScriptInfo(params.textDocument);
+    if (lsInfo === undefined) {
+      return undefined;
+    }
+    const {languageService, scriptInfo} = lsInfo;
+    const offset = lspPositionToTsPosition(scriptInfo, params.position);
+    const ls = languageService as ViewEngineOrIvyLanguageService;
+    if ('getTcb' in ls) {
+      return ls.getTcb(scriptInfo.fileName, offset);
```

Comment:
Typically, the way this was done in the past would be add a stub implementation of `getTcb` in the VE LS that's just `return undefined`. I would prefer that a bit more since it requires less change when we eventually remove the VE version. 
With that, I think you can change the return type of `getLSAndScriptInfo` to be `NgLanguageService`

Reply:
Yeah you're right. We'll update on the angular side.